chain/src/txhashset.rs

```diff
@@ -550,8 +550,13 @@ impl<'a> Extension<'a> {
 		}
 	}
 
-	/// Validate the current txhashset state against a block header
-	pub fn validate(&self, header: &BlockHeader) -> Result<(), Error> {
+	/// Validate the txhashset state against the provided block header.
+	/// Rewinds to that pos for the header first so we see a consistent
+	/// view of the world.
+	pub fn validate(&mut self, header: &BlockHeader) -> Result<(), Error> {
+		// first rewind to the provided header
+		&self.rewind(header)?;
```

Comment:
Good catch. As `validate` is mostly used during a fast sync, this was actually done here for that specific case:

https://github.com/mimblewimble/grin/blob/master/chain/src/chain.rs#L521

Speaking of which, doesn't this break `validate` for fast sync then? Because there shouldn't be any block marker for headers yet. We could have `txhashset_write` in `Chain` save a block marker prior to validation then?

Reply:
Hmm good point - didn't consider fast sync in this...
Let me take a look and see what's involved in saving the block marker earlier.